R/mle_helpers.R

```diff
@@ -0,0 +1,98 @@
+library(lme4)
+library(penalized)
+library(optimx)
+
+fetch_data <- function(data, data_points=NULL, newdata=NULL) {
+  if (is.null(data_points))
+    if (is.null(newdata))
+      return(data)
+    else
+      return(newdata)
+  else if (is.null(newdata))
+    return(data[data_points,, drop = FALSE])
+  else
+    return(newdata[data_points,, drop = FALSE])
+}
+
+linear_mle <- function(form, dat, regul=NULL)
```

Comment:
argument `regul` is there to have the same interface in all of these functions so that the caller does not need to care about it. I'll probably delete the penalised method as we don't really use it.